repository/BaselineOfGrease.package/BaselineOfGrease.class/instance/baselinePharo..st

```diff
@@ -1,39 +1,63 @@
-				package: 'Grease-Tests-Pharo-Core' with: [ spec requires: #('Grease-Tests-Core') ];
-				package: 'Grease-Pharo40-Slime' with: [ spec requires: #('Grease-Core') ];
-				package: 'Grease-Tests-Slime' with: [ spec requires: #('Grease-Pharo40-Slime') ].
-			spec
-				group: 'Slime' with: #('Grease-Pharo40-Slime');
-				group: 'Slime Tests' with: #('Grease-Tests-Slime') ].
+
 	spec
 		for: #(#'pharo4.x' #'pharo5.x')
 		do: [ spec
 				package: 'Grease-Core' with: [ spec includes: #('Grease-Pharo30-Core') ];
+				package: 'Grease-Tests-Pharo-Core' with: [ spec requires: #('Grease-Tests-Core') ];
 				package: 'Grease-Tests-Core'
 					with: [ spec
 						requires: #('Grease-Pharo30-Core');
 						includes: #('Grease-Tests-Pharo-Core') ];
 				package: 'Grease-Pharo30-Core' with: [ spec requires: #('Grease-Core') ] ].
+			
 	spec
 		for: #'pharo6.x'
 		do: [ spec
 				package: 'Grease-Core' with: [ spec includes: #('Grease-Pharo60-Core') ];
+				package: 'Grease-Tests-Pharo-Core' with: [ spec requires: #('Grease-Tests-Core') ];
 				package: 'Grease-Tests-Core'
 					with: [ spec
 						requires: #('Grease-Pharo60-Core');
 						includes: #('Grease-Tests-Pharo-Core') ];
 				package: 'Grease-Pharo60-Core' with: [ spec requires: #('Grease-Core') ] ].
 
 	spec
-		for: #(#'pharo7.x' #'pharo8.x' #'pharo9.x')
+		for: #(#'pharo7.x' #'pharo8.x')
 		do: [ spec
 				package: 'Grease-Core' with: [ spec includes: #('Grease-Pharo70-Core') ];
+				package: 'Grease-Tests-Pharo-Core' with: [ spec requires: #('Grease-Tests-Core') ];
 				package: 'Grease-Tests-Core'
 					with: [ spec
 						requires: #('Grease-Pharo70-Core');
 						includes: #('Grease-Tests-Pharo-Core') ];
-				package: 'Grease-Pharo70-Core' with: [ spec requires: #('Grease-Core') ] ]
\ No newline at end of file
+				package: 'Grease-Pharo70-Core' with: [ spec requires: #('Grease-Core') ] ].
+
+	spec
+		for: #(#'pharo4.x' #'pharo5.x' #'pharo6.x' #'pharo7.x' #'pharo8.x')
+		do: [ 
+			spec
+				package: 'Grease-Pharo40-Slime' with: [ spec requires: #('Grease-Core') ];
+				package: 'Grease-Tests-Slime' with: [ spec requires: #('Grease-Pharo40-Slime') ].
+			spec
+				group: 'Slime' with: #('Grease-Pharo40-Slime');
+				group: 'Slime Tests' with: #('Grease-Tests-Slime') ].
+
+	spec
+		for: #'pharo9.x'
+		do: [ 
+			spec
+				package: 'Grease-Core' with: [ spec includes: #('Grease-Pharo90-Core') ];
+				package: 'Grease-Tests-Pharo-Core' with: [ spec requires: #('Grease-Tests-Core') ];
+				package: 'Grease-Tests-Core'
+					with: [ spec
+						requires: #('Grease-Pharo90-Core');
+						includes: #('Grease-Tests-Pharo-Core') ];
+				package: 'Grease-Pharo90-Core' with: [ spec requires: #('Grease-Core') ];
+				package: 'Grease-Pharo90-Slime' with: [ spec requires: #('Grease-Core') ];
+				package: 'Grease-Tests-Slime' with: [ spec requires: #('Grease-Pharo90-Slime') ].
+
+			spec
+				group: 'Slime' with: #('Grease-Pharo90-Slime');
+				group: 'Slime Tests' with: #('Grease-Tests-Slime') ].
```

Comment:
Why did you make a clone of the entire Slime package? To keep the packages from showing as dirty? AFAICT you only had to change one method because of the deprecation of `#keyword`.

Reply:
We changed the strategy to make a complete package per Pharo version rather than keep shared packages and version-specific packages. It became a mess to understand and maintain in the long run. Also, I anticipate more changes to Slime are necessary because some reports have been that it's broken since a while. 

It's true that this means more work backporting changes.

Reply:
Sounds reasonable. I think that's the right strategy. In Fuel I partially use inheritance and that has become quite messy.